### PR TITLE
Event edit check

### PR DESCRIPTION
**Pull request by chrusher**
Checks whether the sheet columns have changed while editing.

UI is clunky but it works.

Solves #166

### CODE REVIEW

File: thrimbletrimmer/scripts/IO.js [hunk 1]
```diff
@@ -247,20 +248,32 @@ thrimbletrimmerSubmit = function(state) {
...
     console.log(JSON.stringify(wubData));
 
 	if (!wubData.video_start) {alert("No start time set"); return;}
 	if (!wubData.video_end) {alert("No end time set"); return;}
 
     //Submit to thrimshim
-    var rowId = /id=(.*)(?:&|$)/.exec(document.location.search)[1];
-    fetch("/thrimshim/"+rowId, {
+	var rowId = /id=(.*)(?:&|$)/.exec(document.location.search)[1];
+	path = "/thrimshim/" + rowId + "?override_changes=" + override_changes;
```
Comment (ekimekim) [inline thrimbletrimmer/scripts/IO.js:275]: it looks like we're putting override_changes in both the query param and the body. It should be one or the other.

File: thrimshim/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 FROM alpine:3.7
 # dependencies needed for compiling c extensions
 # also busybox-extras for telnet for easier use of backdoor
-RUN apk --update add py2-pip gcc python-dev musl-dev busybox-extras
+RUN apk --update add py2-pip gcc python-dev musl-dev file make busybox-extras
```
Comment (ekimekim) [inline thrimshim/Dockerfile:4]: Rebase your changes to remove these unrelated changes.

File: thrimshim/thrimshim/main.py [hunk 3]
```diff
@@ -184,6 +184,7 @@ def get_row(ident):
 @authenticate
 def update_row(ident, editor=None):
 	new_row = flask.request.json
+        override_changes = 'override_changes' in new_row
```
Comment (ekimekim) [inline thrimshim/thrimshim/main.py:187]: spaces used for indentation here, replace with tabs
```suggestion
	override_changes = 'override_changes' in new_row
```

File: thrimshim/thrimshim/main.py [hunk 4]
```diff
@@ -192,14 +193,16 @@ def update_row(ident, editor=None):
 	non_null_columns = ['upload_location', 'video_start', 'video_end',
 		'video_channel', 'video_quality', 'video_title', 'video_description']
 	edit_columns = non_null_columns + ['allow_holes', 'uploader_whitelist']
+	sheet_columns = ['sheet_name', 'event_start', 'event_end', 'category',
+					 'description', 'notes']
```
Comment (ekimekim) [inline thrimshim/thrimshim/main.py:197]: style nitpick
```suggestion
	sheet_columns = [
		'sheet_name', 'event_start', 'event_end', 'category', 'description', 'notes'
	]
```

File: thrimshim/thrimshim/main.py [hunk 5]
```diff
@@ -218,19 +221,30 @@ def update_row(ident, editor=None):
...
 	#check a row with id = ident is in the database
-	results = database.query(conn, """
-		SELECT id, state 
+	built_query = sql.SQL("""
+		SELECT id, state, {} 
 		FROM events
-		WHERE id = %s""", ident)
-	old_row = results.fetchone()
+		WHERE id = %s""").format(sql.SQL(', ').join(
+			sql.Identifier(key) for key in sheet_columns))
```
Comment (ekimekim) [inline thrimshim/thrimshim/main.py:228]: style nitpick
```suggestion
		WHERE id = %s
	""").format(sql.SQL(', ').join(
		sql.Identifier(key) for key in sheet_columns
	))
```

File: thrimshim/thrimshim/main.py [hunk 6]
```diff
@@ -354,7 +368,7 @@ def stop():
 		# and when not
 		else:
 			sys.exit()
-	gevent.signal(signal.SIGTERM, stop)
+	gevent.signal_handler(signal.SIGTERM, stop)
```
Comment (ekimekim) [inline thrimshim/thrimshim/main.py:371]: Please rebase to remove these unrelated changes.

**Review by ekimekim — COMMENTED** [review]
The mechanism looks good, though I worry a little bit about false positives when comparing the iso-formatted timestamps. We'll see how it goes.

Please rebase to remove the unrelated changes, then add the `tags` column to be checked.

When we do merge this, we'll need to make sure editors understand what it means when it happens.

File: thrimshim/thrimshim/main.py [hunk 7]
```diff
@@ -184,6 +184,7 @@ def get_row(ident):
 @authenticate
 def update_row(ident, editor=None):
 	new_row = flask.request.json
+	override_changes = 'override_changes' in new_row
```
Comment (ekimekim) [inline thrimshim/thrimshim/main.py:187]: this will let us override changes if I set `override_changes: false`, which is surprising.
Let's check it's there and true.
```suggestion
	override_changes = new_row.get('override_changes', False)
```

**Review by ekimekim — APPROVED** [review]
lgtm except one minor issue

**Review by ekimekim — APPROVED** [review]

**Comment by chrusher** [comment]
Resolves #166
